File: python/customer.py
class Customer:
    def __init__(self, name, balance):
        self.name = name
        self.balance = balance
    
    def get_name(self):
        return self.name
    
    def get_balance(self):
        return self.balance
    
    def deduct_balance(self, amount):
        if amount > self.balance:
            raise ValueError("Insufficient balance")
        self.balance -= amount
    
    def has_balance(self, amount):
        return self.balance >= amount

File: cpp/Customer.cpp
#include "Customer.h"
#include <stdexcept>

Customer::Customer(std::string name, double balance)
    : name(name), balance(balance) {}

std::string Customer::getName() const {
    return name;
}

double Customer::getBalance() const {
    return balance;
}

void Customer::deductBalance(double amount) {
    if (amount > balance) {
        throw std::invalid_argument("Insufficient balance");
    }
    balance -= amount;
}

bool Customer::hasBalance(double amount) const {
    return balance >= amount;
}
